bundles/org.openhab.binding.ferroamp/src/main/java/org/openhab/binding/ferroamp/internal/FerroampBindingConstants.java

```diff
@@ -0,0 +1,207 @@
+    public static final String CHANNEL_INVERTERENERGYPRODUCEDL3 = "inverter-energy-produced-l3";
+    public static final String CHANNEL_INVERTERENERGYCONSUMEDL1 = "inverter-energy-consumed-l1";
+    public static final String CHANNEL_INVERTERENERGYCONSUMEDL2 = "inverter-energy-consumed-l2";
+    public static final String CHANNEL_INVERTERENERGYCONSUMEDL3 = "inverter-energy-consumed-l3";
+    public static final String CHANNEL_LOADENERGYPRODUCEDL1 = "load-energy-produced-l1";
+    public static final String CHANNEL_LOADENERGYPRODUCEDL2 = "load-energy-produced-l2";
+    public static final String CHANNEL_LOADENERGYPRODUCEDL3 = "load-energy-produced-l3";
+    public static final String CHANNEL_LOADENERGYCONSUMEDL1 = "load-energy-consumed-l1";
+    public static final String CHANNEL_LOADENERGYCONSUMEDL2 = "load-energy-consumed-l2";
+    public static final String CHANNEL_LOADENERGYCONSUMEDL3 = "load-energy-consumed-l3";
+    public static final String CHANNEL_GRIDENERGYPRODUCEDTOTAL = "grid-energy-produced-total";
+    public static final String CHANNEL_GRIDENERGYCONSUMEDTOTAL = "grid-energy-consumed-total";
+    public static final String CHANNEL_INVERTERENERGYPRODUCEDTOTAL = "inverter-energy-produced-total";
+    public static final String CHANNEL_INVERTERENERGYCONSUMEDTOTAL = "inverter-energy-consumed-total";
+    public static final String CHANNEL_LOADENERGYPRODUCEDTOTAL = "load-energy-produced-total";
+    public static final String CHANNEL_LOADENERGYCONSUMEDTOTAL = "load-energy-consumed-total";
+    public static final String CHANNEL_TOTALSOLARENERGY = "total-solar-energy";
+    public static final String CHANNEL_STATE = "state";
+    public static final String CHANNEL_TIMESTAMP = "timestamp";
+
+    // List of battery setup Channel ids
+    public static final String CHANNEL_BATTERYENERGYPRODUCED = "battery-energy-produced";
+    public static final String CHANNEL_BATTERYENERGYCONSUMED = "battery-energy-consumed";
+    public static final String CHANNEL_SOC = "soc";
+    public static final String CHANNEL_SOH = "soh";
+    public static final String CHANNEL_POWERBATTERY = "power-battery";
+    public static final String CHANNEL_TOTALCAPACITYBATTERIES = "total-capacity-batteries";
+
+    // List of SSO Channel ids
+    public static final String CHANNEL_S1ID = "s1-id";
+    public static final String CHANNEL_S1PVVOLTAGE = "s1-pv-voltage";
+    public static final String CHANNEL_S1PVCURRENT = "s1-pv-current";
+    public static final String CHANNEL_S1TOTALSOLARENERGY = "s1-total-solar-energy";
+    public static final String CHANNEL_S1RELAYSTATUS = "s1-relay-status";
+    public static final String CHANNEL_S1TEMPERATURE = "s1-temperature";
+    public static final String CHANNEL_S1FAULTCODE = "s1-fault-code";
+    public static final String CHANNEL_S1DCLINKVOLTAGE = "s1-dc-link-voltage";
+    public static final String CHANNEL_S1TIMESTAMP = "s1-timestamp";
```

Comment:
Did you consider to use a channel group ? As SSO 1-4 all use the same set of channels, it seems to be a perfect case for a channel group that is re-used.

Reply:
I am not convinced that this will be better. If I will use a channel group so do I have to make 2 new files with, as far I can understand, with the very same info. as it is in thing-types.
But I am not balky in this so if you have some very good ideas I can gladly change mine opinion.